presto-hive/src/main/java/com/facebook/presto/hive/statistics/ParquetQuickStatsBuilder.java

```diff
@@ -0,0 +1,346 @@
+import com.facebook.presto.hive.metastore.Table;
+import com.facebook.presto.parquet.ParquetDataSource;
+import com.facebook.presto.parquet.cache.ParquetFileMetadata;
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.SchemaTableName;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import io.airlift.slice.Slice;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.Path;
+import org.apache.parquet.column.statistics.DoubleStatistics;
+import org.apache.parquet.column.statistics.FloatStatistics;
+import org.apache.parquet.column.statistics.IntStatistics;
+import org.apache.parquet.column.statistics.LongStatistics;
+import org.apache.parquet.column.statistics.Statistics;
+import org.apache.parquet.hadoop.metadata.BlockMetaData;
+import org.apache.parquet.hadoop.metadata.ColumnChunkMetaData;
+import org.apache.parquet.hadoop.metadata.ColumnPath;
+import org.apache.parquet.hadoop.metadata.ParquetMetadata;
+import org.apache.parquet.schema.LogicalTypeAnnotation.DateLogicalTypeAnnotation;
+import org.apache.parquet.schema.LogicalTypeAnnotation.LogicalTypeAnnotationVisitor;
+import org.apache.parquet.schema.LogicalTypeAnnotation.TimeLogicalTypeAnnotation;
+import org.apache.parquet.schema.PrimitiveType;
+
+import java.time.LocalDate;
+import java.time.chrono.ChronoLocalDate;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalLong;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeoutException;
+
+import static com.facebook.presto.hive.CacheQuota.NO_CACHE_CONSTRAINTS;
+import static com.facebook.presto.hive.HivePartition.UNPARTITIONED_ID;
+import static com.facebook.presto.hive.HiveSessionProperties.getReadNullMaskedParquetEncryptedValue;
+import static com.facebook.presto.hive.parquet.HdfsParquetDataSource.buildHdfsParquetDataSource;
+import static com.facebook.presto.hive.parquet.ParquetPageSourceFactory.createDecryptor;
+import static com.facebook.presto.parquet.cache.MetadataReader.readFooter;
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.CompletableFuture.supplyAsync;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+public class ParquetQuickStatsBuilder
+        implements QuickStatsBuilderInterface
+{
+    public static final Logger log = Logger.get(ParquetQuickStatsBuilder.class);
+    private static final ExecutorService footerFetchExecutor = Executors.newCachedThreadPool();
+
+    private static final Set<String> PARQUET_SERDE_CLASS_NAMES = ImmutableSet.<String>builder()
+            .add("org.apache.hadoop.hive.ql.io.parquet.serde.ParquetHiveSerDe")
```

Comment:
Was following the pattern from https://github.com/prestodb/presto/blob/aaf917d7f351a3dbadd8f00a0240979a1a5745b7/presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetPageSourceFactory.java#L153-L156

IMO hardcoding is the better way since class namespaces can get refactored, but existing parquet files will have the serde name still referring to the old one

Reply:
Is there ever a case where this map won't be exactly the same as the one in `ParquetPageSourceFactory`? Can we just expose that set instead of creating a copy?